pkg/registry/apps/daemonset/strategy.go

```diff
@@ -123,7 +123,7 @@ func (daemonSetStrategy) Validate(ctx context.Context, obj runtime.Object) field
 	daemonSet := obj.(*apps.DaemonSet)
 	opts := pod.GetValidationOptionsFromPodTemplate(&daemonSet.Spec.Template, nil)
 	allErrs := validation.ValidateDaemonSet(daemonSet, opts)
-	allErrs = append(allErrs, rest.ValidateDeclaratively(ctx, legacyscheme.Scheme, obj)...)
+	allErrs = append(allErrs, rest.ValidateDeclaratively(ctx, nil, legacyscheme.Scheme, obj)...)
```

Comment:
Currently the opts passed to ValidateDeclaratively are `nil`.  In the future would the nil be modified to be opts from for example:
```go
opts := pod.GetValidationOptionsFromPodTemplate(&daemonSet.Spec.Template, nil)
```

Trying to understand how the available options are parsed from *.GetValidationOptions* methods (or if they are?)

How is the options context `options` field populated with the options that are currently on/off for a given struct?
```
type Context struct {
	Operation Operation
	// Options tracks the options enabled for validation.
	Options sets.Set[string]
}
```

Reply:
Yes, currently the `GetValidationOptionsFrom` functions are hand crafted. The convention is for each to be defined as a struct with (mostly) boolean fields.  I imagine these being either fully replaced with a `Set[string]`, or by adding a converter.  

> How is the options context options field populated with the options that are currently on/off for a given struct?

Presence in the set indicates the option is enabled.  For example, `PodValidationOptions{AllowInvalidPodDeletionCost: true }` could be represented as a string set where `"AllowInvalidPodDeletionCost"` is in the set.

I suppose we could instead offer an interface for options that could be implemented.. that might be handy during migration...  Would that be helpful?

Reply:
Ah got it, thanks.  +1, I think that such an interface would likely be helpful as you mention but definitely not necessary for the PR here.